### DISCUSSION TITLE
forge build version pragma problem?

### DISCUSSION THREAD

**Discussion by aathan** [foundry-rs/foundry#4593] [Help]
EDIT:  The below problem may be because solidity 0.8.20 is not in fact a valid release? I assumed it was because currently, searches for solidity information yield results such as:
```
[Using the Compiler — Solidity 0.8.20 documentation](https://docs.soliditylang.org/en/latest/using-the-compiler.html)
One of the build targets of the Solidity repository is solc, the solidity commandline compiler. Using solc --help provides you with an explanation of all options. The compiler can produce various outputs, ranging from simple binaries and assembly over an abstract syntax tree (parse tree) to estimations of
```

But in fact, 0.8.20 is not listed at https://github.com/ethereum/solidity/releases

So, perhaps forge needs to improve its error message when there are version pragmas that do not resolve to valid compiler versions??
Filed as #4594

----------------

I think there may be a problem with ethers_solc::resolver, possibly having to do with lexical vs numerical comparisons of the version numbers; or perhaps with how the dot versions are parsed??  Simply changing the version pragma in XYZPausableProxy.sol to anything that starts with a 1, for example ^0.8.19 no longer generates the incompatible solidity version issue.

Should I open a bug or am I making an error?

```
$ forge --version
forge 0.2.0 (06f7c9c 2023-03-18T00:08:48.50351195Z)
$ forge build --force
[⠃] Compiling...2023-03-18T21:42:22.476642Z ERROR ethers_solc::resolver: failed to resolve versions
[⠊] Compiling...
Error: 
Discovered incompatible solidity versions in following
: src/XYZPausableProxy.sol (^0.8.20) imports:
    lib/openzeppelin-contracts/contracts/proxy/transparent/ProxyAdmin.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/StorageSlot.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/access/Ownable.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/proxy/ERC1967/ERC1967Proxy.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/Context.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/proxy/Proxy.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/proxy/ERC1967/ERC1967Upgrade.sol (^0.8.2)
    lib/openzeppelin-contracts/contracts/proxy/beacon/IBeacon.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/interfaces/draft-IERC1822.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/Address.sol (^0.8.1)
    lib/openzeppelin-contracts/contracts/utils/StorageSlot.sol (^0.8.0)
```

**Comment by mds1** [comment]
Is this still an issue? I saw you also opened https://github.com/foundry-rs/foundry/issues/4594 which has been resolved. From the trace you posted it it seems the problem is that `XYZPausableProxy` specifies `^0.8.20` which is not released yet
